toolchain/install/busybox_info_test.cpp

```diff
@@ -166,5 +167,58 @@ TEST_F(BusyboxInfoTest, NotBusyboxSymlink) {
   EXPECT_FALSE(info.ok());
 }
 
+TEST_F(BusyboxInfoTest, LayerSymlinksInstallTree) {
+  auto actual_busybox = MakeFile(dir_ / "actual-busybox");
+
+  // Create a facsimile of the install prefix with even the busybox as a
+  // symlink. Also include potential relative sibling symlinks like `clang++` to
+  // `clang`.
+  auto prefix = dir_ / "test_prefix";
+  MakeDir(prefix);
+  MakeDir(prefix / "lib");
+  MakeDir(prefix / "lib/carbon");
+  MakeSymlink(prefix / "lib/carbon/carbon-busybox", actual_busybox);
+  MakeDir(prefix / "lib/llvm");
+  MakeDir(prefix / "lib/llvm/bin");
+  auto clangplusplus_target =
+      MakeSymlink(prefix / "lib/llvm/bin/clang++", "clang");
+  auto clang_target =
+      MakeSymlink(prefix / "lib/llvm/bin/clang", "../../carbon-busybox");
+  MakeDir(prefix / "bin");
+  auto carbon_target =
+      MakeSymlink(prefix / "bin/carbon", "../lib/carbon/carbon-busybox");
+
+  auto info = GetBusyboxInfo(carbon_target.string());
+  ASSERT_TRUE(info.ok()) << info.error();
+  EXPECT_TRUE(info->bin_path.is_absolute());
+  EXPECT_THAT(info->mode, Eq(std::nullopt));
+
+  info = GetBusyboxInfo(clang_target.string());
+  ASSERT_TRUE(info.ok()) << info.error();
+  EXPECT_TRUE(info->bin_path.is_absolute());
+  EXPECT_THAT(info->mode, Eq("clang"));
+
+  info = GetBusyboxInfo(clangplusplus_target.string());
+  ASSERT_TRUE(info.ok()) << info.error();
+  EXPECT_TRUE(info->bin_path.is_absolute());
```

Comment:
`is_absolute` isn't going to test much; a return value of "/bin/carbon" would satisfy this check. Is this test doing enough that you're confident this is the _correct_ absolute path?